web-app/src/main/webapp/script.js

```diff
@@ -93,24 +94,42 @@ function showMessageOnInfoWindow(message, position, map, infoWindow) {
   infoWindow.open(map);
 }
 
-function showReportForm() {
+function showReportForm(map, geocoder) {
   document.getElementById('form-container').style.display = 'block';
   const homeElements = document.getElementsByClassName('home');
   for (const element of homeElements) {
     element.style.display = 'none';
   }
+
+  geocoder.geocode({ 'location': map.getCenter() }, (results, status) => {
+    if (status === 'OK') {
+      if (results[0]) {
+        document.getElementById('location-input').value = results[0].formatted_address;
+      } else {
+        console.error('No results found');
+      }
+    } else {
+      console.error('Geocoder failed due to: ' + status);
+    }
+  })
 }
 
-function hideReportForm() {
+function hideReportForm(reset) {
   document.getElementById('form-container').style.display = 'none';
   const homeElements = document.getElementsByClassName('home');
   for (const element of homeElements) {
     element.style.display = 'block';
   }
+
+  if (reset) {
+    document.getElementById('report-form').reset();
+  }
```

Comment:
This doesn't really belong inside hideReportForm, I think it would be better if you just put this code inside the lambda i.e.

```
addEventListener('click', () => {
  hideReportForm();
  if (reset) {
    document.getElementById('report-form').reset();
  }
});
```